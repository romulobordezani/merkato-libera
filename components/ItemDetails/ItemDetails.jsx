import PropTypes from 'prop-types';
import React, { Fragment } from 'react';

import styles from './ItemDetails.scss';
import Price from '../Price';

const ItemDetails = ({ item }) => {
  return (
    <Fragment>
      <div className={styles.ItemDetails_imgBox}>
        <img src={item.picture} alt={item.title} />
      </div>

      <div className={styles.ItemDetails_infoBox}>
        <div className={styles.ItemDetails_infoBox_info}>
          {item.condition === 'new' ? 'Nuevo' : 'Usado'} - {item.sold_quantity} vendido
          {item.sold_quantity > 1 && 's'}
        </div>
        <div className={styles.ItemDetails_infoBox_title}>{item.title}</div>
        <Price item={item} fontSize="46px" />
        <a href={item.permalink} className={styles.ItemDetails_infoBox_buyButton}>
          Comprar
        </a>
      </div>

      <div className={styles.ItemDetails_descriptionBox}>
        <p className={styles.ItemDetails_descriptionBox_title}>Descripci√≥n del producto</p>
        <p className={styles.ItemDetails_descriptionBox_content}>{item.description}</p>
      </div>
    </Fragment>
  );
};

ItemDetails.propTypes = {
  item: PropTypes.object.isRequired
};

export default ItemDetails;
