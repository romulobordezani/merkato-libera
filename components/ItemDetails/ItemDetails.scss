@import "../_SassLib/breakpoints";

.ItemDetails {

  &_imgBox {
    padding-top: 32px;
    grid-area: imagBox;
    display: flex;
    align-content: center;
    justify-content: center;

    img {
      width: 100%;
      height: auto;
      object-fit: contain;
      border-radius: 4px !important;

      @include breakpoint($large) {
        width: 680px;
        max-height: 680px;
      }

    }
  }

  &_infoBox {
    grid-area: infoBox;
    padding: 32px 32px 0 0;

    @include breakpoint($small) {
      padding: 0;
    }

    &_info {
      font-size: 14px;
    }

    &_title {
      padding-top: 14px;
      font-size: 24px;
      font-weight: 500;
    }



    &_buyButton {
      display: inline-block;
      margin-right: 10px;
      background: #3483FA;
      color: #fff;
      text-decoration: none;
      line-height: 38px;
      border-radius: 4px;
      -webkit-transition: all 0.3s;
      transition: all 0.3s;
      text-align: center;
      cursor: pointer;
      font-size: 18px;
      width: 100%;

      &:hover {
        background: lighten(#3483FA, 10%);
      }
    }

    &_freeShipping {
      outline: none;
      margin-left: 6px;
    }
  }

  &_descriptionBox {
    grid-area: descBox;
    padding: 0 32px 32px 32px;

    @include breakpoint($small) {
      padding: 0;
    }

    &_title {
      font-size: 28px;
      margin-bottom: 32px;
      font-weight: 400;
    }

    &_content {
      white-space: pre-line;
      line-height: 20px;
      font-size: 16px;
    }
  }
}
