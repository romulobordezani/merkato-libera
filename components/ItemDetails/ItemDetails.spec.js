import React from 'react';
import renderer from 'react-test-renderer';

import ItemDetails from './ItemDetails';

const ItemMock = require('../../__mocks__/Item');

describe('ItemDetails', () => {
  it('Snapshot renders default', () => {
    expect(renderer.create(<ItemDetails item={ItemMock} />)).toMatchSnapshot();
  });
});
