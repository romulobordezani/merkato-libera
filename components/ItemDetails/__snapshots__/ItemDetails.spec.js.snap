// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails Snapshot renders default 1`] = `
Array [
  <div>
    <img
      alt="Botella Deportiva Térmica Reutilizable Pura Vita 500 Ml"
      src="https://mla-s1-p.mlstatic.com/943145-MLA31085352819_062019-O.jpg"
    />
  </div>,
  <div>
    <div>
      Nuevo
       - 
      420
       vendido
      s
    </div>
    <div>
      Botella Deportiva Térmica Reutilizable Pura Vita 500 Ml
    </div>
    <span
      style={
        Object {
          "fontSize": "46px",
        }
      }
    >
      <span>
        $
      </span>
      <span>
        1.800
      </span>
      <span>
        
      </span>
    </span>
    <a
      href="https://articulo.mercadolibre.com.ar/MLA-694674962-botella-deportiva-termica-reutilizable-pura-vita-500-ml-_JM"
    >
      Comprar
    </a>
  </div>,
  <div>
    <p>
      Descripción del producto
    </p>
    <p>
      »Tienda Oficial«
• Pura • Cuidamos tu Agua. 
Somos Empresa B certificada. Queremos cambiar el Mundo a través del Agua y estamos convencidos que podemos hacerlo.

IMPORTANTE: Este producto se despacha desde el depósito de MercadoLibre, y sólo disponen de los colores indicados en la publicación.
Si quisieras 2 unidades o más de diferentes colores debes agregar cada color deseado al carrito de compras. 
No podemos cambiar los colores seleccionados una vez hecha la compra. La mensajería interna se habilitará una vez efectuada la entrega. 

Botella Térmica 500 ml 24hs Frío / 12hs Calor.

Marca: Pura.
Modelo: Vita.
Capacidad: 500 ml.
Medidas: 27cm x 6,6 cm (alto/diámetro)
Material: Acero inoxidable grado alimenticio 18/8.
Tapa: 2 tapas; a rosca metálica y pico con sorbete.
Peso: 272 g.
Presentación: En tubo de cartón.

Importante: Pueden variar levemente los colores a la fotografía expuesta en la publicación. 

Botella deportiva Pura 500 ml
Con Pura podés expresar tu personalidad eligiendo la botella que mejor se adapta a tu estilo, entre una amplia gama de exclusivos colores.
Las botellas Pura son ideales para llevar tu bebida Pura, fría o caliente, vayas donde vayas. Te permiten beber agua de calidad en cualquier lugar y en cualquier momento del día, convirtiéndose en tu complemento hidratación inseparable.

* Están fabricadas con un material tan brillante como su diseño, liviano, resistente y duradero. 
* Son térmicas, por lo que mantiene tu bebida fría o caliente por más de 12 horas estimado.
* Al ser aislantes no traspasa la temperatura interior al exterior.
* Son ecológicas y reutilizables.
* Se lavan fácilmente, incluso en el lavavajillas y su superficie especial no permite la adherencia de hongos, detritos ni restos de ninguna naturaleza.
* No desprenden olores ni sabores.
* Libres de BPA (Bisfenol A).

Tu próxima Pura Vita ya te está esperando.

¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨
» QUIENES SOMOS:
Nuestra misión es desarrollar tecnologías que generen en las personas salud y bienestar a través del agua, cuidando el medio ambiente.
Creemos que todo cambio positivo puede ser posible y que empieza por nosotros mismos. Queremos que cuando una persona vea, escuche, sienta Pura, se conecte con una mejor calidad de vida y contagie a otros.

» FACTURACIÓN:
Realizamos factura A y B. 

» ENVÍOS:
Envío gratis, a domicilio o sucursal, en compras a partir de $1999 desde el depósito de MercadoEnvíos Full. 


¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨

Esto es por los océanos, los ríos y lagos, para que sigan siendo nuestra fuente de vida. Esto es por nuestro planeta, para que lo podamos disfrutar y admirar. Esto es por las personas, para que sigamos sintiendo, soñando, creando y compartiendo. Esto es por amor al agua, el origen de la naturaleza, nuestro alimento y nuestra fuente de inspiración. Esto es por nosotros, los que creemos que debemos ser el cambio que queremos ver en el mundo, los que queremos dejarle a nuestros hijos, un mejor lugar donde vivir. Los que creemos que podemos elegir una forma diferente de vivir la vida, en armonía y con respeto. Los que estamos convencidos de que este mundo no es mejor, si todos no estamos mejor. Los que creemos en que cuidar el mundo es la clave para soñar, vivir, reir y amar.

Cuando amamos, cuidamos. Y cuando cuidamos, amamos. Be Pura. Be the Change.
    </p>
  </div>,
]
`;
